src/Components/Components/src/NavigationManager.cs

```diff
@@ -203,7 +205,15 @@ public virtual void Refresh(bool forceReload = false)
 
     private void NotFoundCore()
     {
-        _notFound?.Invoke(this, new NotFoundEventArgs());
+        if (_notFound == null)
+        {
+            // global router doesn't exist, no events were registered
+            NavigateTo($"{BaseUri}not-found");
+        }
```

Comment:
We shouldn't trigger a navigation to a fixed URL, the behavior should be to no-op if there aren't any handlers.

Reply:
I'm pretty sure that we decided that without global router, initially we're navigating to fixed path.

Reply:
I don't think we understood each other. I only meant that for the JS code that deals with enhanced nav/streaming, not for code on the server.